server/proto.go

```diff
@@ -37,14 +42,12 @@ func writeMessage(conn net.Conn, msgs []*message.Signed) error {
 		fmt.Printf("%s\n", jsonData)
 	}
 
-	_, err = conn.Write(frameMessage(reply))
+	// Extend the deadline, we just sent a message.
+	err = conn.SetDeadline(time.Now().Add(deadline))
```

Comment:
git made this misleading. The `return err` below is new logic attached to `SetDeadline` now. Is it ok for failed deadline to return instead of ignoring it?

Reply:
Go docs are super spartan on what the hell it can return as an error.

I guess it's a good idea to bail outta here if error.. sure.

Reply:
This turned out to be an important issue. A user that sends a message and then disconnects induces a race condition here where we could get the message for sending and then drop the message just because we can't set the deadline.

I removed the returns for both send+disconnect and receive+disconnect so that we still deal with the messages regardless of the connection status.